src/test/ui/intrinsics/panic-uninitialized-zeroed.rs

```diff
@@ -70,21 +67,42 @@ enum ZeroIsValid {
 }
 
 #[track_caller]
-fn test_panic_msg<T>(op: impl (FnOnce() -> T) + panic::UnwindSafe, msg: &str) {
-    let err = panic::catch_unwind(op).err();
-    assert_eq!(
-        err.as_ref().and_then(|a| a.downcast_ref::<&str>()),
-        Some(&msg)
-    );
+fn test_panic_msg<T, F: (FnOnce() -> T) + 'static>(op: F, msg: &str) {
+    use std::{panic, env, process};
+
+    // The tricky part is that we can't just run `op`, as that would *abort* the process.
+    // So instead, we reinvoke this process with the caller location as argument.
+    // For the purpose of this test, the line number is unique enough.
+    // If we are running in such a re-invocation, we skip all the tests *except* for the one with that type name.
+    let our_loc = panic::Location::caller().line().to_string();
+    let mut args = env::args();
+    let this = args.next().unwrap();
+    if let Some(type_name) = args.next() {
+        if type_name == our_loc {
+            op();
+            panic!("we did not abort");
+        } else {
+            // Nothing, we are running another test.
+        }
+    } else {
+        // Invoke new process for actual test, and check result.
+        let mut cmd = process::Command::new(this);
+        cmd.arg(our_loc);
+        let res = cmd.output().unwrap();
+        assert!(!res.status.success(), "test did not fail");
+        let stderr = String::from_utf8_lossy(&res.stderr);
+        assert!(stderr.contains(msg), "test did not contain expected output: looking for {:?}, output:\n{}", msg, stderr);
+    }
```

Comment:
This does make the test very slow. But that seems fine? I can't think of a better way...

Reply:
You could spawn all the processes and only once that's done collect all the results. But that requires a bit more plumbing.

Reply:
Or these tests could be rewritten as `#[test]` functions. This is also used in some other UI tests.
And lib-test spawns new processes for tests when the it's compiled with `panic=abort` or some unstable flag is passed. Not sure if compiletest can pass arguments to run-pass tests.

Reply:
You have to use `-Cpanic=abort` and `-Zpanic-abort-tests` together when compiling the crate to be tested to have libtest spawn new processes.

Reply:
Hm, yeah I guess that could work, though I would have to restructure the test quite a bit.

Let's see what the reviewer says, for now I'd prefer this approach.

Reply:
I think this approach looks okay. It launches 75 processes or so, which I don't think will be too much time in the grand scheme of things. It might be worse on Windows since process creation is more expensive there though. Given that we run the whole test suite in parallel, I doubt one slower test will affect the overall time too much.